python/lsst/cp/pipe/utilsEfd.py

```diff
@@ -257,41 +315,163 @@ def parseMonochromatorStatus(self, data, dateStr):
+            The table of results returned from the EFD.
+        """
+        # All electrometer data gets written to the same series.
+        defaultSeries = "lsst.sal.Electrometer.logevent_intensity"
+        alternateSeries = "lsst.sal.Electrometer.logevent_logMessage"
+        dataSeries = dataSeries if dataSeries else defaultSeries
 
-            if myTime <= dateValue:
-                low = idx
-            elif myTime > dateValue:
-                high = idx
+        if dateMin:
+            startDate = Time(dateMin, format='isot', scale='tai')
+        else:
+            startDate = None
+        if dateMax:
+            stopDate = Time(dateMax, format='isot', scale='tai')
+        else:
+            stopDate = None
 
-            idx = (high + low) // 2
-            iteration += 1
-            if high - low == 1:
-                found = True
-            self.log.debug("parse search %d %d %d %d %s %s",
-                           low, high, idx, found, myTime, dateValue)
+        if (dataSeries == alternateSeries and (startDate is None or stopDate is None)):
+            raise RuntimeError("Cannot query logevent_logMessage without dates to limit memory issues.")
 
-        # End binary search.
+        results = self.selectTimeSeries(dataSeries, [],
+                                        startDate, stopDate)
+        if dataSeries == 'lsst.sal.Electrometer.logevent_logMessage':
+            results = self.rewriteElectrometerStatus(results)
+        return results
+
+    def rewriteElectrometerStatus(self, inResults):
+        """Rewrite intermediate electrometer data from the EFD.
+
+        Parameters
+        ----------
+        inResults : `astropy.table.Table`
+            The table of results returned from the EFD.
+
+        Returns
+        -------
+        outResults : `astropy.table.Table`
+            The rewritten table containing only electrometer summary
+            status events.
+        """
+        # This is fragile against upstream changes.
+        # Ignore all entries that are not the ones we care about.
+        outResults = inResults[np.where(inResults['functionName'] == 'write_fits_file')]
+        outResults = outResults[np.where(outResults['level'] == 20)]
+
+        # These will be new columns
+        intensityMean = []
+        intensityStdev = []
+        intensityTime = []
+        intensityFile = []
+
+        for row in outResults:
+            # Fallback values in case the regexp fails
+            mean = np.nan
+            stdev = np.nan
+            time_mean = np.nan
+            filename = "REGEXP_FAIL"
+
+            # Find the last "filename\.extension" before the first newline;
+            #      the last [] grouped values before the second newline;
+            #      the last [] grouped values before the end of the string.
+            magic = re.findall(r'\b\w+.+?(\w+?\.\w+)\n\b\w+.+\[(.*?)\]\n\b\w+.+\[(.*?)\]$',
+                               row['message'])
```

Comment:
oh wow this is ... not ideal but I understand why you did it.

Reply:
Yeah.  I'm hoping we can deprecate this function as quickly as possible.